File: Easy/Python/392. Is Subsequence.py
class Solution(object):
    def isSubsequence(self, s, t):
        if len(s) == 0:
            return True
        if len(t) < len(s):
            return False
        
        sub = 0

        for i in range(len(t)):
            if s[sub] == t[i]:
                sub += 1
                if sub == len(s):
                    return True

        return False

File: Easy/JavaScript/392. Is Subsequence.js
var isSubsequence = function(s, t) {
    if(s.length == 0)
        return true;
    if(t.length < s.length)
        return false;
    
    let sub = 0;

    for(let i = 0; i < t.length; i++){
        if(s[sub] === t[i])
            sub++;
        if(sub === s.length)
            return true;
    }

    return false;
};
